src/coreclr/vm/comcallablewrapper.cpp

```diff
@@ -3315,7 +3315,8 @@ void ComMethodTable::LayOutClassMethodTable()
 
                 if (pMD &&
                         !(pCurrInteropMD ? IsDuplicateClassItfMD(pCurrInteropMD, i) : IsDuplicateClassItfMD(pMD, i)) &&
-                    IsOverloadedComVisibleMember(pMD, pParentMD))
+                    IsOverloadedComVisibleMember(pMD, pParentMD) &&
+                    !pMD->IsAsyncMethod())
```

Comment:
```suggestion
if (pMD
    && !(pCurrInteropMD ? IsDuplicateClassItfMD(pCurrInteropMD, i) : IsDuplicateClassItfMD(pMD, i))
    && IsOverloadedComVisibleMember(pMD, pParentMD)
    && !pMD->IsAsyncMethod())
```

This was really difficult to parse.